staging-next: another iteration

vcunat opened the pull request:
This might be **the last** iteration before forking 19.03, so let me /cc @lheckemann.

FRidh committed:
buildPython*: enable strictDeps

timokau committed:
python.pkgs.entrypoints: simplify tests

The upstream issue requiring one test to be disabled was fixed:
https://github.com/takluyver/entrypoints/issues/23

Izorkin committed:
pythonPackages.psutil: 5.4.8 -> 5.5.1

dtzWill committed:
gtk3: 3.24.4 -> 3.24.5

* add fribidi as it's now required.
* fetch patches to fix fribidi linking, autoreconf

Ma27 committed:
buildEnv: break with a proper error if one path is actually a file

I noticed by creating `buildEnv` where I accidentally put a derivation
from `pkgs.writeText` into `paths` and got a broken build with the
following misleading error message:

```
Use of uninitialized value $stat1 in numeric ne (!=) at /nix/store/9g4wc31j7a2xp22xpgwr0qssfxahxdzl-builder.pl line 74.
Use of uninitialized value $stat1 in bitwise and (&) at /nix/store/9g4wc31j7a2xp22xpgwr0qssfxahxdzl-builder.pl line 75.
different permissions in `' and `/nix/store/0vy5ss91laxvwkyvrbld5hv27i88qk5w-noise': 0000 <-> 0444 at /nix/store/9g4wc31j7a2xp22xpgwr0qssfxahxdzl-builder.pl line 75.
```

It can be reproduced with an expression like this:

``` nix
{ pkgs ? import <nixpkgs> { } }:
let
  file = pkgs.writeText "test" ''
    content
  '';
in
  pkgs.buildEnv {
    name = "test-env";
    paths = [ /* ... */ file ];
  }
```

catern committed:
pythonPackages.pkgconfig: use same setupHook as pkgconfig

This setupHook is necessary for actually using pkgconfig. Without this
setupHook, you need to put both pythonPackages.pkgconfig and pkgconfig
into your buildInputs, just to get the setuphook of the latter.

Previously, pythonPackages.pkgconfig depended on pkgconfig as a
propagatedBuildInput, so this setupHook was propagated down to
dependents. That was changed to a regular nativeBuildInput, so now
this setupHook change is necessary for packages to only depend on
pythonPackages.pkgconfig.

A possible alternative is extracting out a common pkgconfigHook
package and having both pythonPackages.pkgconfig and pkgconfig depend
on that with propagatedBuildInputs. That might be necessary to avoid
the buildHook running twice?

Commit:
python.pkgs.python-services: init at 1.3.1

Commit:
python3.pkgs.cherrypy: 18.0.1 -> 18.1.0

FRidh committed:
python: trezor_agent: 0.9.3 -> 0.10.0

FRidh committed:
python: tox: 3.5.3 -> 3.7.0

FRidh committed:
python: traits: 4.6.0 -> 5.0.0

FRidh committed:
python: trustme: 0.4.0 -> 0.5.0

FRidh committed:
python: twine: 1.12.1 -> 1.13.0

FRidh committed:
python: trio: 0.10.0 -> 0.11.0

FRidh committed:
python: thespian: 3.9.5 -> 3.9.7

FRidh committed:
python: tqdm: 4.28.1 -> 4.31.1

FRidh committed:
python: tifffile: 0.15.1 -> 2019.2.10

FRidh committed:
python: twitter.common.options: 0.3.9 -> 0.3.10

FRidh committed:
python: txtorcon: 18.3.0 -> 19.0.0

FRidh committed:
python: typed-ast: 1.1.2 -> 1.3.1

FRidh committed:
python: uproot: 3.2.15 -> 3.4.5

FRidh committed:
python: twitter.common.log: 0.3.9 -> 0.3.10

FRidh committed:
python: uproot-methods: 0.2.11 -> 0.4.2

FRidh committed:
python: twitter.common.confluence: 0.3.9 -> 0.3.10

FRidh committed:
python: vidstab: 1.0.1 -> 1.5.6

FRidh committed:
python: vega_datasets: 0.5.0 -> 0.7.0

FRidh committed:
python: twitter.common.collections: 0.3.9 -> 0.3.10

FRidh committed:
python: twitter.common.lang: 0.3.9 -> 0.3.10

FRidh committed:
python: twitter.common.dirutil: 0.3.9 -> 0.3.10

FRidh committed:
python: venusian: 1.1.0 -> 1.2.0

FRidh committed:
python: waitress: 1.1.0 -> 1.2.1

FRidh committed:
python: wheel: 0.32.3 -> 0.33.0

FRidh committed:
python: virtualenv: 16.1.0 -> 16.4.0

FRidh committed:
python: xkcdpass: 1.16.5 -> 1.17.2

FRidh committed:
python: xlrd: 1.1.0 -> 1.2.0

FRidh committed:
python: yapf: 0.25.0 -> 0.26.0

FRidh committed:
python: virtualenv-clone: 0.2.6 -> 0.5.1

FRidh committed:
python: wsproto: 0.12.0 -> 0.13.0

FRidh committed:
python: wrapt: 1.10.11 -> 1.11.1

FRidh committed:
python: xmltodict: 0.11.0 -> 0.12.0

FRidh committed:
python: voluptuous-serialize: 2.0.0 -> 2.1.0

FRidh committed:
python: virtualenvwrapper: 4.8.2 -> 4.8.4

FRidh committed:
python: xml2rfc: 2.12.3 -> 2.18.0

FRidh committed:
python: yamllint: 1.14.0 -> 1.15.0

FRidh committed:
python: ZConfig: 3.3.0 -> 3.4.0

FRidh committed:
python: zope.deprecation: 4.3.0 -> 4.4.0

FRidh committed:
python: zm-py: 0.3.0 -> 0.3.3

FRidh committed:
python: yattag: 1.10.1 -> 1.11.1

FRidh committed:
python: zope.configuration: 4.3.0 -> 4.3.1

FRidh committed:
python.pkgs.bootstrapped-pip: wheel 0.32.2 -> 0.33.0, setuptools 40.6.3 -> 40.8.0, pip 18.1 -> 19.0.2

FRidh committed:
python: pytest: 3.9.3 -> 4.2.1

FRidh committed:
python.pkgs.freezegun: use correct inputs

FRidh committed:
python: pycodestyle: 2.4.0 -> 2.5.0

FRidh committed:
python.pkgs.flake8: 3.5.0 -> 3.7.5

FRidh committed:
python.pkgs.html5lib: disable certain tests

due to pytest change

Commit:
python.pkgs.pyflakes: run tests

FRidh committed:
python.pkgs.pytest_3: keep older version

because not all test suites are compatible with 4 yet.

FRidh committed:
python.pkgs.soupsieve: init at 1.7.3

Commit:
python.pkgs.pycodestyle: remove unused argument

FRidh committed:
python.pkgs.pytest-cov: remove unused deps

FRidh committed:
python.pkgs.ipykernel: disable failing tests

FRidh committed:
python.pkgs.characteristic: fix pytest 4 incompatibility

Commit:
python.pkgs.whoosh: use pytest_3

With pytest_42, we get
`[pytest] section in setup.cfg files is no longer supported`.

Commit:
borgbackup: use msgpack-python-0.5.6

Commit:
python.pkgs.parameterized: update dependencies

Commit:
python.pkgs.pytest-timeout: fix tests

Commit:
python.pkgs.Mako: use pytest_3

With pytest_42, we get
`[pytest] section in setup.cfg files is no longer supported`.

Commit:
python.pkgs.txaio: use pytest_3

Otherwise, tests fail with
`Fixtures are not meant to be called directly, but are created
automatically when test functions request them as parameters.`

Commit:
python.pkgs.requests_toolbelt: disable tests that access network

Commit:
python.pkgs.requests_toolbelt: add meta.license

Commit:
home-assistant: fix pinned dependencies' builds

Commit:
python.pkgs.pytest-localserver: fix tests

Commit:
python.pkgs.hkdf: no tests in tarball

Commit:
python.pkgs.PasteDeploy: no tests in tarball

Commit:
python.pkgs.aiodns: add typing to propagatedBuildInputs

Commit:
python.pkgs.virtualenv: rebase virtualenv-change-prefix.patch

Commit:
python.pkgs.oauthlib: 2.0.0 -> 3.0.1

Commit:
python.pkgs.praw: use pytest_3

FRidh committed:
awscli: 1.16.90 -> 1.16.104

FRidh committed:
dosage: fix build

FRidh committed:
Revert "python.pkgs.matplotlib: use pkgs.pkgconfig"

This reverts commit c78f98db56b580bada5c56ef38793f4fa5c140b6.

FRidh committed:
vulnix: ronn is a native build input

FRidh committed:
matplotlib: fix pkgconfig situation

We need pkgs.pkgconfig, and not pythonPackages.pkgconfig.

Commit:
python.pkgs.pylibgen: no tests in tarball

Commit:
papis: fix build with recent python-slugify

Commit:
python.pkgs.mutagen: fix tests with updated pycodestyle

Commit:
python.pkgs.portend: disable flake8 tests

They fail with many occurences of `E117 over-indented`.

Commit:
python.pkgs.billiard: add psutil to propagatedBuildInputs

Commit:
python.pkgs.kombu: add Pyro4 to checkInputs

FRidh committed:
python.pkgs.protobuf: fix build

FRidh committed:
python.pkgs.pyqt5: fix build

FRidh committed:
python.pkgs.qtconsole: distinguish between build and check inputs

FRidh committed:
python.pkgs.cypari2: fix build

after setting `strictDeps = true;`

FRidh committed:
python.pkgs.murmurhash: fix build

FRidh committed:
Python: fix outfall after setting `strictDeps = true;`

FRidh committed:
python.pkgs.partd: disable failing test

FRidh committed:
python.pkgs.datashape: mark as broken

FRidh committed:
python.pkgs.wurlitzer: init at 1.0.2

FRidh committed:
spyder: 3.2.8 -> 3.3.3

FRidh committed:
python.pkgs.qtawesome: fix build

FRidh committed:
python.pkgs.distro: mark as broken

FRidh committed:
python.pkgs.trollius: only for python 2

FRidh committed:
python.pkgs.pythonix: mark as broken

FRidh committed:
buildPythonPackage: always export `LANG=C.UTF-8`

FRidh committed:
pants: mark as broken

FRidh committed:
python.pkgs: rename pytestflakes to pytest-flakes

to follow naming guideline

FRidh committed:
python: Wand: 0.5.0 -> 0.5.1

FRidh committed:
python: WazeRouteCalculator: 0.7 -> 0.7.2

FRidh committed:
python: pytest-forked: 1.0.1 -> 1.0.2

FRidh committed:
python: google-cloud-dataproc: 0.3.0 -> 0.3.1

FRidh committed:
python: llfuse: 1.3.5 -> 1.3.6

FRidh committed:
python: pyupdate: 0.2.23 -> 0.2.26

FRidh committed:
python: gin-config: 0.1.2 -> 0.1.3

FRidh committed:
python: pid: 2.2.2 -> 2.2.3

FRidh committed:
python: boto3: 1.9.94 -> 1.9.96

FRidh committed:
python: httplib2: 0.12.0 -> 0.12.1

FRidh committed:
python: cairocffi: 1.0.1 -> 1.0.2

FRidh committed:
python: PyChromecast: 2.5.0 -> 2.5.1

FRidh committed:
python: alerta-server: 6.7.3 -> 6.7.4

FRidh committed:
python: pyro-ppl: 0.3.1 -> 0.3.1.post1

FRidh committed:
python: botocore: 1.12.94 -> 1.12.96

FRidh committed:
python: identify: 1.2.1 -> 1.2.2

FRidh committed:
python: marshmallow: 2.18.0 -> 2.18.1

FRidh committed:
python: sopel: 6.6.2 -> 6.6.3

FRidh committed:
python: texttable: 1.6.0 -> 1.6.1

FRidh committed:
python: telethon-session-sqlalchemy: 0.2.7 -> 0.2.8

FRidh committed:
awscli: 1.16.104 -> 1.16.106

FRidh committed:
arelle: fix build

FRidh committed:
Revert "python: XStatic-Bootstrap: 3.3.7.1 -> 4.1.3.1"

because this broke the sole user bepasty

This reverts commit c7b4424d58a3d091fd62281829d740349a067851.

FRidh committed:
octoprint: 1.3.9 -> 1.3.10, still broken

need to add a bunch of custom versions of dependencies

FRidh committed:
python.pkgs.alerta: fix build

FRidh committed:
python.pkgs.alerta-server: fix build

FRidh committed:
python.pkgs.altair: fix build

FRidh committed:
buildPythonPackage: initial support for PEP 517

ikervagyok committed:
xorg/libXau: 1.0.8 -> 1.0.9

ikervagyok committed:
xorg/xcalc: init ad 1.0.7

ikervagyok committed:
xorg/xcursorgen: 1.0.6 -> 1.0.7

ikervagyok committed:
xorg/xcursor-themes: 1.0.5 -> 1.0.6

ikervagyok committed:
xorg/xf86-video-chips: 1.3.4 -> 1.4.0

ikervagyok committed:
xorg/xf86-video-s3virge: 1.10.7 -> 1.11.0

ikervagyok committed:
xorg/xf86-video-tdfx: 1.4.7 -> 1.5.0

ikervagyok committed:
mesa: update llvmPackagages 6 -> 7

ikervagyok committed:
xorg/xf86-video-apm: init at 1.3.0

ikervagyok committed:
mesa: 18.3.3 -> 18.3.4

illegalprime committed:
cmake: add ability to cross-compile cmake

matthewbauer committed:
Merge remote-tracking branch 'NixOS/master' into staging

matthewbauer committed:
Revert "cctools: bump to latest commit"

This reverts commit ac682e362c07929b2b30bfdf4f75cd1c5250d3e7.

This broke iOS building on master. Even Xcode 8.2 comes with TAPI
librarises. We need these patches to support those .tbd files.
Eventually we will move to using libtapi directly, but I have not
finished work on this right now.

Unfortunately, this will not have my changes for building cctools with
manpages. We will have to do this update at some later time.

matthewbauer committed:
cctools: add myself as maintainer

matthewbauer committed:
Revert "libcxx: add linker scripts for libc++.a to ensure libc++abi.a is properly linked"

This reverts commit 72e1764199805ab6e602fa3762a4d95bfdf80cb8.

This causes the GHC panic reported in issue #55848.

vcunat committed:
Merge #55981: gtk3: 3.24.4 -> 3.24.5 (into staging)

vcunat committed:
Merge branch 'staging-next' into staging

vcunat committed:
libpng: always apply the debian patches

This fixes the low-severity CVE on other platforms than aarch64.

matthewbauer committed:
Merge pull request #56029 from matthewbauer/revert-cctools-bump

Revert cctools bump

matthewbauer committed:
Merge pull request #56030 from matthewbauer/revert-72e176

Revert "libcxx: add linker scripts for libc++.a to ensure libc++abi.a…

Commit:
bump cmake to 3.13.4

matthewbauer committed:
Merge pull request #56021 from illegalprime/fix/cmake-cross

cmake: add ability to cross-compile cmake

matthewbauer committed:
Merge pull request #54993 from yorickvP/patch-2

wayland: separateDebugInfo = true

Ma27 committed:
buildEnv: improve file check to avoid false-positives

The original change in #55372 was supposed to fix the case where a store
path which is a file should be placed into `buildEnv` which broke with a
fairly misleading Perl error.

Unfortunately this introduced a regression, `findFiles` can have targets
that are files if the file isn't a store path. Rather than adding more
obscure checks with probably further regressions, I figured that it's
better to replicate the behavior of `lib.isStorePath` and explicitly
check if the store path is a file and break in this case only.

This should also fix recent staging issues.

Commit:
python.pkgs.mwclient: use checkInputs

Commit:
python.pkgs.git-annex-adapter: use patch to hardcode git-annex path

Commit:
python.pkgs.celery: 4.2.1 -> 4.3.0rc1

Using a pre-release because it currently does not build due to
updated dependencies.

Commit:
python.pkgs.paster-pastedeploy: fix tests compatibility with PasteDeploy 2+

Commit:
python.pkgs.readthedocs-sphinx-ext: init at 0.5.16

Commit:
python.pkgs.sphinx_rtd_theme: fix tests

Commit:
python.pkgs.soco: use pytest_3

Otherwise, we get
    [pytest] section in setup.cfg files is no longer supported,
    change to [tool:pytest] instead

Commit:
beets: use checkInputs

Commit:
python.pkgs.pygame: fix build

after setting `strictDeps = true;`

Commit:
python.pkgs.fb-re2: no tests in tarball

Commit:
octoprint: ignore version constraints

Commit:
octoprint: fix tests

dtzWill committed:
file: 5.35 -> 5.36

https://mailman.astron.com/pipermail/file/2019-February/000073.html

FRidh committed:
Merge staging-next into staging

FRidh committed:
Merge staging into python-unstable

FRidh committed:
Merge pull request #55757 from NixOS/python-unstable

Python: package set updates before branch-off

7c6f434c committed:
Merge pull request #55764 from oxij/pkgs/nasm-from-source

nasm: build from source

dtzWill committed:
Merge pull request #56126 from dtzWill/update/file-5.36

file: 5.35 -> 5.36

dtzWill committed:
Revert "file: 5.35 -> 5.36"

vcunat committed:
Merge #56022: mesa and x11 updates (into staging)

vcunat committed:
Merge #56170: Revert "file: 5.35 -> 5.36" (staging)

vcunat committed:
Merge #56059: cmake: 3.12.1 -> 3.13.4 (into staging)

vcunat committed:
Merge #55632: libuv: 1.24.1 -> 1.26.0 (into staging)

vcunat committed:
Merge #55613: libidn2: 2.1.0 -> 2.1.1 (into staging)

vcunat committed:
Merge #55819: unbound: 1.8.3 -> 1.9.0 (into staging)

It would be quite a shame if 19.03 missed dnsflagday.net changes
and some other features/fixes from 1.9.0.

dtzWill committed:
nghttp2: 1.35.1 -> 1.36.0

https://github.com/nghttp2/nghttp2/releases/tag/v1.36.0

Commit:
python: readthedocs-sphinx-ext: 0.5.16 -> 0.5.17

vcunat committed:
Merge #55779: only set __darwinAllowLocalNetworking on darwin

... into staging.

vcunat committed:
Merge #55420: buildEnv, substituteAll: disable binary cache

... into staging.

vcunat committed:
Merge branch 'staging' into staging-next

vcunat committed:
Merge branch 'master' into staging-next

Hydra nixpkgs: ?compare=1506218

FRidh committed:
python.pkgs.hypothesis: 3.79.3 -> 3.88.3

FRidh committed:
python.pkgs.configparser: fix darwin regression

We can use C.UTF-8 on Linux but not yet on Darwin.

vcunat commented:
@FRidh: something seems wrong about python packages and their build inputs.  Many of them now fail to find things &ndash; in $PATH or includes or just dependency detection.  Most of [new failures](https://hydra.nixos.org/eval/1506333?filter=x86_64-linux&compare=1506218&full=) seem to be like that.

FRidh commented:
@vcunat this is because I set `strictDeps = true;` in `buildPythonPackage`. Many were fixed but not all. The remainder can be fixed on `master`.

vcunat commented:
I see, it should certainly help cross-compilation and probably clean up overall over longer term.  I'm not sure this period immediately before forking off stable is a good moment, as probably many fixes will need to be cherry-picked.

FRidh committed:
python: aiohue: 1.9.0 -> 1.9.1

FRidh committed:
python: XlsxWriter: 1.1.4 -> 1.1.5

FRidh committed:
python: boto3: 1.9.96 -> 1.9.101

FRidh committed:
python: cx_Oracle: 7.1.0 -> 7.1.1

FRidh committed:
python: androguard: 3.3.4 -> 3.3.5

FRidh committed:
python: configparser: 3.7.1 -> 3.7.3

FRidh committed:
python: atom: 0.4.2 -> 0.4.3

FRidh committed:
python: django-contrib-comments: 1.9.0 -> 1.9.1

FRidh committed:
python: cmd2: 0.9.8 -> 0.9.10

FRidh committed:
python: alerta-server: 6.7.4 -> 6.7.5

FRidh committed:
python: botocore: 1.12.96 -> 1.12.101

FRidh committed:
python: cffi: 1.12.0 -> 1.12.1

FRidh committed:
python: flake8: 3.7.5 -> 3.7.6

FRidh committed:
python: google-auth: 1.6.2 -> 1.6.3

FRidh committed:
python: pyrsistent: 0.14.10 -> 0.14.11

FRidh committed:
python: ledgerblue: 0.1.22 -> 0.1.23

FRidh committed:
python: phonenumbers: 8.10.5 -> 8.10.6

FRidh committed:
python: PyChromecast: 2.5.1 -> 2.5.2

FRidh committed:
python: murmurhash: 1.0.1 -> 1.0.2

FRidh committed:
python: pykeepass: 3.0.2 -> 3.0.3

FRidh committed:
python: plaid-python: 2.4.0 -> 2.4.1

FRidh committed:
python: prompt_toolkit: 2.0.8 -> 2.0.9

FRidh committed:
python: pyupdate: 0.2.26 -> 0.2.29

FRidh committed:
python: py_stringmatching: 0.4.0 -> 0.4.1

FRidh committed:
python: pyhomematic: 0.1.55 -> 0.1.56

FRidh committed:
python: globus-sdk: 1.7.0 -> 1.7.1

FRidh committed:
python: pip: 19.0.2 -> 19.0.3

FRidh committed:
python: pytest-server-fixtures: 1.6.1 -> 1.6.2

FRidh committed:
python: pyzmq: 17.1.2 -> 17.1.3

FRidh committed:
python: qiskit: 0.7.2 -> 0.7.3

FRidh committed:
python: uproot: 3.4.5 -> 3.4.6

FRidh committed:
python: uproot-methods: 0.4.2 -> 0.4.3

FRidh committed:
python: zope.i18n: 4.6.1 -> 4.6.2

FRidh committed:
python: regex: 2019.02.07 -> 2019.02.21

FRidh committed:
python: quantities: 0.12.2 -> 0.12.3

FRidh committed:
python: wheel: 0.33.0 -> 0.33.1

FRidh committed:
python: spectral-cube: 0.4.3 -> 0.4.4

FRidh committed:
python: telethon-session-sqlalchemy: 0.2.8 -> 0.2.9.post1

FRidh committed:
python: virtualenv: 16.4.0 -> 16.4.1

FRidh committed:
python.pkgs.bootstrapped-pip: pip 19.0.2 -> 19.0.3, wheel 0.33.0 -> 0.33.1

FRidh committed:
python.pkgs.rhpl: fix build

catern committed:
pythonPackages.pykerberos: take krb5 as a nativeBuildInput (#56268)

LnL7 committed:
cmake: disable ApplicationServices and CoreServices on darwin

LnL7 committed:
xar: add explicit lzma input

lzma is part of the stdenv but not during bootstrapping?

FRidh committed:
Merge master into staging-next

LnL7 committed:
cmake: add cf-private on darwin

vcunat committed:
Merge #56291: cmake: fix darwin bootstrapping

... into staging-next.

costrouc committed:
pythonPackages.howdoi: refactor fix HOME requirement

costrouc committed:
pythonPackages.gidgethub: refactor build without flit

costrouc committed:
pythonPackages.monty: 1.0.2 -> 1.0.4

and cleanup broken tests

costrouc committed:
pythonPackages.pynacl: refactor remove deprecated hypothesis testing args

costrouc committed:
pythonPackages.pint: refactor change pypi name and 27 compatability

Commit:
python.pkgs.fiona: fix build

Commit:
vdirsyncer: fix tests

Commit:
python.pkgs.cherrypy: skip test that fails intermittently

See https://github.com/cherrypy/cherrypy/issues/1306 for the
upstream discussion.

dotlambda commented on pkgs/development/python-modules/fiona/default.nix:
```diff
@@ -14,24 +15,35 @@ buildPythonPackage rec {
   CXXFLAGS = stdenv.lib.optionalString stdenv.cc.isClang "-std=c++11";
 
+  nativeBuildInputs = [
+    gdal # for gdal-config
```
@FRidh Just wanted to let you know that I believe this is a situation where a package needs to be in `nativeBuildInputs` as well as `buildInputs`.

FRidh replied:
Probably yes. `gdal-config` requires it to be in `nativeBuildInputs`, but we also build against it so also in `buildInputs`.

FRidh commented:
Python failure of `jupyter`/`nbconvert` is transient.

Commit:
pythonPackages.shapely: use patch to set library paths

Commit:
python.pkgs.pynacl: do not propagate libsodium

dotlambda committed:
Merge pull request #56312 from costrouc/python-unstable

Fixes several broken python packages for python-unstable

lheckemann commented:
This won't magically be ready in the next hour, will it? :/ I guess we'll have to rebase it into the release branch after the fact. :(

FRidh commented:
I think this can go in. Aarch64 is currently broken, but that can be fixed up.

alyssais committed:
openssl: 1.0.2q -> 1.0.2r

(cherry picked from commit 9c94d7483681cf00bcffa15a6f47845d2be09b68)
From PR #56414.  Forward-picking from staging to staging-next,
as minor openssl bumps tend to be fairly breakage-free.

FRidh commented:
@vcunat @lheckemann how to proceed to get this into `staging-19.03`?

FRidh commented:
I've merged staging-next into both master and staging-19.03.
